### DISCUSSION TITLE
SwiftGodot Discussions are now available on Matrix!

### DISCUSSION THREAD

**Discussion by alicerunsonfedora** [migueldeicaza/SwiftGodot#350] [Announcements]
> [!IMPORTANT]
> Matrix is provided as an **additional option** to chat with others in the workspace. You are _not_ required to use Matrix to join us. If you are unfamiliar with Matrix or see no reason to use it, you are more than welcome to join us from the Slack workspace.

After some discussions and discovery, I've gone ahead and bridged a good portion of our Slack workspace to Matrix! This should allow us to maintain an archive of our discussions going forward, and extend the community to a wider audience that would prefer not to use Slack.
Some details to note about this bridging:
- The main community space is hosted at `#swiftgodot:one.ems.host`. You can join that to gain access to all its related rooms. I don't remember if joining this way is automatic, but I know this space is helpful for organization. Think of it as the entry point like a Discord server or this very workspace.
- Because bridging is on a per-channel basis, I've covered the five most popular channels:
  - The [#general](https://swiftongodot.slack.com/archives/C05CD0SFZ6X) chat is hosted at `#swiftgodot-general:one.ems.host`.
  - The [#swiftgodot](https://swiftongodot.slack.com/archives/C05BYJ95S5D) channel is hosted at `#swiftgodot-discussion:one.ems.host`.
  - The [#swiftgodotkit](https://swiftongodot.slack.com/archives/C05CA6KB1QV) channel is hosted at `#swiftgodot-swiftgodotkit:one.ems.host`.
  - The [#random](https://swiftongodot.slack.com/archives/C05CFH51GDA) channel is hosted at `#swiftgodot-random:one.ems.host`.
  - The [#showcase](https://swiftongodot.slack.com/archives/C05CFKXPUUU) channel is hosted at `#swiftgodot-showcase:one.ems.host`.
- Per above, I will add/bridge more channels when necessary, should they become more active.
- Finally, I'd like to ask some community members from Slack to join the Matrix community and help moderate content. When I last did such bridging for a Discord server, we had plenty of bad actors and spammers come from the Matrix side to post very inappropriate content that didn't align with our guidelines, thus prompting to shut down the integration. I'd love to keep these bridges going, so I would appreciate extra help in making sure everything is okay coming from both sides.

**Comment by alicerunsonfedora** [comment]
Since the introduction of the Matrix bridging, I've bridged over a couple of new rooms:

- #matrix is hosted at `#swiftgodot-matrix:one.ems.host`.
- #visionpro-integration is hosted at `#swiftgodot-visionpro:one.ems.host`

**Reply by daniele-niero** [reply]
Hi,

> _and extend the community to a wider audience that would prefer not to use Slack_

I get why some people don't want to use Slack, starting from the annoying thing that discussions older than 90 days (I think) are locked for non paying users.

However, I quickly tried to access this Matrix chat and I have to say, I'm even more reluctant. I just finished to read what is Matrix, if it is safe, who are the people behind it, will it be stable etc etc... and now I have to chose a client... I don't want to go through all that process again and I don't want to install something on my devices without knowing what I'm putting on them.

Why using this tool and not, for example, Discord?
Isn't there the risk to push more people away instead of attracting them?

I, for one, I'm not really sure I want to get on board on this Matrix thing. Perhaps it is just me...

(edit)
I'm not suggesting Matrix is bad and Discord is good. I'm not an expert on these tools and technologies, so my post should be read on that light: People not familiar with all the details, caveats and implementation details, might be pushed away when asking them to use something that is not well established and unfamiliar to them. 
For example I tried to join #swiftgodot:one.ems.host using Element on web... I can't figure out how, it keep saying it can't find that room. I tried all sort of different combination without success

**Reply by alicerunsonfedora** [reply]
Hey @daniele-niero! You are not required to use the Matrix side if you want to chat; we merely offer Matrix as an additional option for those who prefer to use it over Slack.

With this in mind, I will try to address some of the points listed here:

> Why using this tool and not, for example, Discord? Isn't there the risk to push more people away instead of attracting them?

To my knowledge, there are no tools that exist to allow two-way communication between Slack and Discord without using Matrix as a bridge. While using Discord is fine with some, not everyone is on board with using Discord. The company often makes questionable decisions that make it harder to use.

> However, I quickly tried to access this Matrix chat and I have to say, I'm even more reluctant. I just finished to read what is Matrix, if it is safe, who are the people behind it, will it be stable etc etc... and now I have to chose a client... I don't want to go through all that process again and I don't want to install something on my devices without knowing what I'm putting on them.

Again, *you are not required to use Matrix to chat with us*. We provide it as an additional option that happens to let us have a more full archive than what Slack offers on free tiers.

> For example I tried to join #swiftgodot:one.ems.host using Element on web... I can't figure out how, it keep saying it can't find that room. I tried all sort of different combination without success

This is a sort of known issue with Element. While you can join _rooms_ easily, spaces seem to be very hit or miss. Try joining an individual room listed above instead.

**Comment by daniele-niero** [comment]
Hi Marquis,

Thank you for explaining. It wasn't clear to me that Slack and Matrix were
working together, I thought Matrix was supposed to be a replacement for
Slack.
It makes more sense now.

Thanks again. Regards
